### PR TITLE
(GH-7) Start lang server without wait

### PR DESCRIPTION
**Pull request by jpogran**
This PR starts the ruby language server without using a delay or sleep when the extension starts, so there is no need to start it yourself

- [x] Start ruby language server async
- [x] Find ruby language server path dynamically
- [ ] Find where Puppet is dynamically
- [ ] Config file setting for puppet path

### CODE REVIEW

File: client/src/languageserver.ts [hunk 1]
```diff
@@ -3,11 +3,54 @@ import {
...
+  var serverPath = path.join(extensionPath, 'server');
+  // "/some/file/structure/vscode-puppet/server/puppet-languageserver"
+  var serverExe = path.join(serverPath, 'puppet-languageserver');
+
+  let cmd = '';
+  let args = [serverExe];
+  let options = {};
+
+  if (process.platform == 'win32') {
+    cmd = 'c:\\windows\\system32\\cmd.exe'
```
Comment (glennsarti) [inline client/src/languageserver.ts]: Need to change this to sysnative on 64bit.  Vscode is a 32bit processs on Windows.  Should probably use %COMSPEC%  too?

**Comment by jpogran** [comment]
This leaves off windows for the moment until we can figure out process spawn correctly there. This also leaves off packaging, as it's now not working again. Need to research more, but both these issues are out of scope for this PR

File: client/src/extension.ts [hunk 2]
```diff
@@ -1,30 +1,61 @@
...
 
-  createStatusBarItem();
+    let config = vscode.workspace.getConfiguration('puppet');
 
-  var languageServerClient = startLangServerTCP(host, port, langID, statusBarItem);
-  context.subscriptions.push(languageServerClient.start());
+    var host             = config['languageserver']['address']; // '127.0.0.1';
+    var port             = config['languageserver']['port']; // 8081;
+    var stopOnClientExit = config['languageserver']['stopOnClientExit']; // true;
+    var timeout          = config['languageserver']['timeout']; // 8081;
```
Comment (glennsarti) [inline client/src/extension.ts]: nit: It's not 8081 for the timeout

File: client/src/languageserver.ts [hunk 3]
```diff
@@ -1,14 +1,53 @@
...
 import * as cp from 'child_process';
 import ChildProcess = cp.ChildProcess;
+import {
+  LanguageClient, LanguageClientOptions, ServerOptions
+} from 'vscode-languageclient'
 
 import * as messages from '../src/messages';
 
-export function startLangServerTCP(host: string, port: number, langID: string, statusBarItem): LanguageClient {
+export function createLanguageServerProcess(serverExe: string, myOutputChannel: vscode.OutputChannel) {
```
Comment (glennsarti) [inline client/src/languageserver.ts]: nit: should it be called Exe?  that's more a Windows-ism

Comment (jpogran) [inline client/src/languageserver.ts]: what should it be called? binary? file?

File: client/src/extension.ts (hunk 2 above)
Comment (jpogran) [inline client/src/extension.ts]: copy-pasta comments

**Pull request merged**